pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -223,3 +227,110 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 		By("Test successful !")
 	})
 }
+
+func getNextAvailableTkrVersion(tkgctlClient tkgctl.TKGClient, currentTkrVersion string) (string, error) {
+	var foundVersion string
+
+	tkrs, err := tkgctlClient.GetTanzuKubernetesReleases("")
+	if err != nil {
+		return "", err
+	}
+
+	for i := range tkrs {
+		if !isCompatible(tkrs[i]) {
+			continue
+		}
+
+		if _, exists := tkrs[i].Labels[runv1alpha3.LabelDeactivated]; exists {
+			continue
+		}
+
+		specVersionIsNewer, err := isNewerVMwareVersion(tkrs[i].Spec.Version, currentTkrVersion)
+		if err != nil {
+			return "", err
+		}
+		if specVersionIsNewer {
+			// if we don't already have a foundVersion we take spec.version
+			foundVersion, err = pickOlderVMwareVersion(foundVersion, tkrs[i].Spec.Version)
+			if err != nil {
+				return "", nil
+			}
+		}
+	}
+
+	if foundVersion == "" {
+		return "", fmt.Errorf("no TKR version available for upgrade")
+	}
+
+	return foundVersion, nil
+}
+
+func isCompatible(tkr v1alpha1.TanzuKubernetesRelease) bool {
+	var compatible string
+	for _, condition := range tkr.Status.Conditions {
+		if condition.Type == runv1alpha3.ConditionCompatible {
+			compatible = string(condition.Status)
+			break
+		}
+	}
+	if !strings.EqualFold(compatible, "true") {
+		return false
+	}
+	return true
+}
+
+func isNewerVMwareVersion(thisVersion, thanVersion string) (bool, error) {
```

Comment:
The name ```thanVersion``` is a bit confusing. How about ```otherVersion```, or just ```tkrVersionA``` and ```tkrVersionB``` similar to your other function?

Reply:
otherVersion and tkrVersionA will be just as confusing :)
Lets leave as is.